deployments/helm/hephaestus/values.yaml

```diff
@@ -1,106 +1,172 @@
+
+  # Extra labels for manager pods
+  podLabels: {}
+
+  # Annotations for manager pods
+  podAnnotations: {}
 
-prometheus:
-  # Enable prometheus scraping
+  # Tolerations for manager pod assignment
+  tolerations: []
+
+  # Affinities for manager pod assignment
+  affinity: {}
+
+  # Node labels for manager pod assignment
+  nodeSelector: {}
+
+  # Manager pods priorityClassName
+  priorityClassName: ""
+
+  # Number of manager instances to run. Leader election will be enabled
+  # whenever this value is greater than 1; this ensures that only one instance
+  # is active at a time.
+  replicaCount: 1
+
+  # Service account configuration
+  serviceAccount:
+    # Specifies whether a service account should be created
+    create: true
+    # Annotations to add to the service account
+    annotations: {}
+    # The name of the service account to use.
+    # If not set and create is true, a name is generated using the fullname template
+    name: ""
+
+# Log processor configuration
+logProcessor:
+  # Log processor image
+  image:
+    registry: ""
+    repository: timberio/vector
+    tag: 0.20.X-distroless-static
+    pullPolicy: IfNotPresent
+
+  # Enable the deployment of the log processing sidecar
   enabled: true
-  # Allow ingress traffic from prometheus pods with the following labels
-  podLabels:
-    app.kubernetes.io/name: prometheus
-  # Namespace labels where prometheus is running if different from app's namespace
-  namespaceLabels: {}
-
-image:
-  registry: ghcr.io
-  repository: dominodatalab/hephaestus
-  pullPolicy: IfNotPresent
-  # Overrides the image tag whose default is the chart appVersion
-  tag: ""
-
-imagePullSecrets: []
-
-serviceAccount:
-  # Specifies whether a service account should be created
-  create: true
-  # Annotations to add to the service account
-  annotations: {}
-  # The name of the service account to use.
-  # If not set and create is true, a name is generated using the fullname template
-  name: ""
-
-podLabels: {}
-podAnnotations: {}
-podSecurityContext: {}
-  # fsGroup: 2000
-
-securityContext: {}
-  # capabilities:
-  #   drop:
-  #   - ALL
-  # readOnlyRootFilesystem: true
-  # runAsNonRoot: true
-  # runAsUser: 1000
-
-resources: {}
-  # We usually recommend not to specify default resources and to leave this as a conscious
-  # choice for the user. This also increases chances charts run on environments with little
-  # resources, such as Minikube. If you do want to specify resources, uncomment the following
-  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
-  # limits:
-  #   cpu: 100m
-  #   memory: 128Mi
-  # requests:
-  #   cpu: 100m
-  #   memory: 128Mi
-
-
-nodeSelector: {}
-
-tolerations: []
-
-affinity: {}
-
-priorityClassName: ""
 
-nameOverride: ""
-fullnameOverride: ""
+  # Resource requests and limits
+  resources:
+    requests: {}
+    limits: {}
+
+  # Configure the ingestion, transformation, and delivery of your logs
+  config:
```

Comment:
It does feel a tiny bit brittle if we're asking e.g. our deployments to configure this with the helm templating included, but I might need a more concrete example of how we want to configure sinks before balking at it. For example, maybe we'll never need to template `sinks` and we're better off with it being actual structured input?

Reply:
My thinking is that both the S3 and Redis sinks have a number of advanced configuration options available that I don't want to constrain nor map out at a higher level. Additionally, other sinks seems interesting (Kafka, NewRelic, etc...) and it might be worth leaving the `sink` config free-form so that it can be extended w/o modifications to the helm chart. That was my reasoning, thoughts?

https://vector.dev/docs/reference/configuration/sinks/vector/